src/update-function/move-array-control.spec.ts

```diff
@@ -0,0 +1,23 @@
+import { createFormArrayState } from '../state';
+import { moveArrayControl } from './move-array-control';
+import { FORM_CONTROL_ID } from './test-util';
+
+describe('moveArrayControl', () => {
+  const INITIAL_ARRAY_STATE = createFormArrayState(FORM_CONTROL_ID, [ 0, 1, 2, 3 ]);
+
+  it('should call reducer for arrays', () => {
+    const resultState = moveArrayControl<number>(1, 0)(INITIAL_ARRAY_STATE);
```

Comment:
With the changes I suggest in the implementation below the explicit generic parameter becomes superfluous. Please remove it. Same for other occurrences in this file.